src/Umbraco.Community.Contentment/DataEditors/SocialLinks/social-links.js

```diff
@@ -0,0 +1,200 @@
+    "overlayService",
+    "Umbraco.Community.Contentment.Services.DevMode",
+    function ($scope, editorService, localizationService, overlayService, devModeService) {
+
+        //console.log("social-links.model", $scope.model);
+
+        var defaultConfig = {
+            confirmRemoval: 0,
+            defaultIcon: "icon-add",
+            defaultBackgroundColor: "#f6f4f4",
+            defaultIconColor: "#9e9e9e",
+            enableDevMode: 0,
+            maxItems: 0,
+            overlayView: "",
+            networks: [],
+        };
+        var config = Object.assign({}, defaultConfig, $scope.model.config);
+
+        var vm = this;
+
+        function init() {
+
+            $scope.model.value = $scope.model.value || [];
+
+            if (Number.isInteger(config.maxItems) === false) {
+                config.maxItems = Number.parseInt(config.maxItems) || defaultConfig.maxItems;
+            }
+
+            config.confirmRemoval = Object.toBoolean(config.confirmRemoval);
+
+            vm.allowAdd = config.maxItems === 0 || $scope.model.value.length < config.maxItems;
+            vm.focusName = false;
+
+            vm.sortableOptions = {
+                axis: "y",
+                containment: "parent",
+                cursor: "move",
+                opacity: 0.7,
+                scroll: true,
+                tolerance: "pointer",
+                stop: function (e, ui) {
+                    setDirty();
+                }
+            };
+
+            config.lookup = _.indexBy(config.networks, "network");
+
+            vm.add = add;
+            vm.edit = edit;
+            vm.getIcon = getIcon;
+            vm.getStyle = getStyle;
+            vm.pick = pick;
+            vm.remove = remove;
+
+            if (Object.toBoolean(config.enableDevMode) === true && $scope.umbProperty) {
+                $scope.umbProperty.setPropertyActions([{
+                    labelKey: "contentment_editRawValue",
+                    icon: "brackets",
+                    method: edit
+                }, {
+                    labelKey: "clipboard_labelForRemoveAllEntries",
+                    icon: "trash",
+                    method: function () {
+                        $scope.model.value = [];
+                    }
+                }]);
+            }
+        };
+
+        function add() {
+
+            vm.focusName = false;
+
+            $scope.model.value.push({
+                network: "",
+                name: "",
+                url: "",
+            });
+
+            pick($scope.model.value.length - 1);
+
+            setDirty();
+        };
+
+        function edit() {
+            devModeService.editValue($scope.model, function () {
+                // NOTE: Any future validation can be done here.
+            });
+        };
+
+        function getIcon(network) {
+
+            var item = config.lookup[network];
+
+            if (item && item.icon) {
+                return item.icon.split(" ")[0];
```

Comment:
I'd like to see if the `icon` value can be sanitised on saving the data-type, rather than needing to manipulate it in the property-editor controller every time.